### DISCUSSION TITLE
need help

### DISCUSSION THREAD

**Discussion by yeyedatou** [xxyzz/WordDumb#126] [General]
Starting job: 正在为《Harry Potter and the Goblet of Fire (Book 4)》生成生词提示和X-Ray 
Job: "正在为《Harry Potter and the Goblet of Fire (Book 4)》生成生词提示和X-Ray" failed with error: 
Traceback (most recent call last):
  File "calibre\gui2\threaded_jobs.py", line 82, in start_work
  File "calibre_plugins.worddumb.parse_job", line 152, in do_job
  File "calibre_plugins.worddumb.deps", line 36, in install_deps
  File "calibre_plugins.worddumb.deps", line 81, in which_python
  File "calibre_plugins.worddumb.utils", line 51, in run_subprocess
  File "subprocess.py", line 524, in run
subprocess.CalledProcessError: Command '['python', '-c', 'import platform; print(".".join(platform.python_version_tuple()[:2]))']' returned non-zero exit status 9009.
 
Called with args: ((15, 'AZW3', 'D:\\Calibre Portable\\Calibre Library\\J.K. Rowling\\Harry Potter and the Goblet of Fire (15)\\Harry Potter and the Goblet of - J.K. Rowling.azw3', <calibre.ebooks.metadata.book.base.Metadata object at 0x000002BBBF434790>, 'en'), True, True) {'notifications': <queue.Queue object at 0x000002BBBF437E20>, 'abort': <threading.Event object at 0x000002BBBF4344F0>, 'log': <calibre.utils.logging.GUILog object at 0x000002BBBF437C70>}

**Comment by xxyzz** [comment]
Please copy the full error message.

I guess you didn't read the document(https://xxyzz.github.io/WordDumb/installation.html) and didn't install Python.

**Reply by dmy72** [reply]
Hi, what should I do with Python after installing Python? Thank you.

**Reply by xxyzz** [reply]
You don't have to do anything with Python after installing it, the plugin will find Python in your system's PATH.

**Reply by dmy72** [reply]
> You don't have to do anything with Python after installing it, the plugin will find Python in your system's PATH.

It shows me this and I have installed Python:

Starting job: 正在为《Harry Potter and the Philosopher's Stone》生成生词提示和X-Ray 
Job: "正在为《Harry Potter and the Philosopher's Stone》生成生词提示和X-Ray" failed with error: 
Traceback (most recent call last):
  File "calibre\gui2\threaded_jobs.py", line 82, in start_work
  File "calibre_plugins.worddumb.parse_job", line 190, in do_job
  File "calibre_plugins.worddumb.utils", line 51, in run_subprocess
  File "subprocess.py", line 524, in run
subprocess.CalledProcessError: Command '['py', 'C:\\Users\\dmy72\\AppData\\Roaming\\calibre\\plugins\\WordDumb.zip', '{"create_ww": true, "create_x": true, "asin": "B019PIOJYU", "book_path": "E:\\\\Calibre \\u4e66\\u5e93\\\\J.K. Rowling\\\\Harry Potter and the Philosopher\'s S (4)\\\\Harry Potter and the Philosophe - J.K. Rowling.azw3", "acr": "CR!7EYZ7ZY3FD24S45PFWKG438BRCGR", "revision": "d159fc17", "model": "en_core_web_md", "lemma_lang": "en", "mobi_codec": "utf-8", "useragent": "WordDumb/3.29.2 (https://github.com/xxyzz/WordDumb)", "book_fmt": "AZW3", "plugin_path": "C:\\\\Users\\\\dmy72\\\\AppData\\\\Roaming\\\\calibre\\\\plugins\\\\WordDumb.zip"}', '{"use_pos": true, "search_people": true, "model_size": "md", "zh_wiki_variant": "cn", "fandom": "", "add_locator_map": true, "preferred_formats": ["KFX", "AZW3", "AZW", "MOBI", "EPUB"], "use_all_formats": false, "minimal_x_ray_count": 1, "en_ipa": "ga_ipa", "zh_ipa": "pinyin", "choose_format_manually": true, "wiktionary_gloss_lang": "zh_cn", "kindle_gloss_lang": "zh_cn", "use_gpu": false, "cuda": "cu118", "last_opened_kindle_lemmas_language": "en", "last_opened_wiktionary_lemmas_language": "en", "use_wiktionary_for_kindle": false, "ca_wiktionary_difficulty_limit": 5, "da_wiktionary_difficulty_limit": 5, "de_wiktionary_difficulty_limit": 5, "el_wiktionary_difficulty_limit": 5, "en_wiktionary_difficulty_limit": 5, "es_wiktionary_difficulty_limit": 5, "fi_wiktionary_difficulty_limit": 5, "fr_wiktionary_difficulty_limit": 5, "hr_wiktionary_difficulty_limit": 5, "it_wiktionary_difficulty_limit": 5, "ja_wiktionary_difficulty_limit": 5, "ko_wiktionary_difficulty_limit": 5, "lt_wiktionary_difficulty_limit": 5, "mk_wiktionary_difficulty_limit": 5, "nl_wiktionary_difficulty_limit": 5, "no_wiktionary_difficulty_limit": 5, "pl_wiktionary_difficulty_limit": 5, "pt_wiktionary_difficulty_limit": 5, "ro_wiktionary_difficulty_limit": 5, "ru_wiktionary_difficulty_limit": 5, "sv_wiktionary_difficulty_limit": 5, "uk_wiktionary_difficulty_limit": 5, "zh_wiktionary_difficulty_limit": 5}']' returned non-zero exit status 1.
 
Called with args: ((4, 'AZW3', "E:\\Calibre 书库\\J.K. Rowling\\Harry Potter and the Philosopher's S (4)\\Harry Potter and the Philosophe - J.K. Rowling.azw3", , 'en'), True, True) {'notifications': , 'abort': , 'log': }

**Reply by xxyzz** [reply]
Please copy the full error message.

**Reply by dmy72** [reply]
> Please copy the full error message.

Above is the full error message that Calibre shows me.

**Reply by xxyzz** [reply]
The error message from the subprocess is a the bottom, that's the real error. The traceback log you posted tells nothing about the error in subprocess.
